sisl/io/sile.py

```diff
@@ -546,6 +547,89 @@ def pre_open(self, *args, **kwargs):
     return _wrapper
 
 
+def sile_read_multiple(start=0, stop=1000000000, step=None, all=False,
+                       skip_call=None, pre_call=None, post_call=None):
+    """ Method decorator for doing multiple reads
+
+    Parameters
+    ----------
+    start : int, optional
+       position of first read
+    stop : int, optional
+       position of last read
+    step : int, optional
+       number of steps to take (if any at all)
+    all : bool, optional
+       read all entries (takes precedence over start/step)
+    skip_call : function, optional
+       read method without actual data processing
+    pre_call : function, optional
+       function to be applied before each read call
+    post_call : function, optional
+       function to be applied after each read call
+
+    Returns
+    -------
+    single entry or list from multiple reads
+    """
+    def decorator(func):
+        @wraps(func)
+        def multiple(self, *args, start=start, stop=stop, step=step, all=all,
```

Comment:
as per your animation file, we should probably force the extra arguments to be forced keys:
```python
def multiple(self, *args, *, start=start, stop=stop, step=step, all=all)
```

Secondly, the `*_call` methods should not be added to the method it self as far as I can see, they could just be used without allowing users to change the behaviour. I don't know if they could be useful for users. Hmm... Perhaps they could... What do you think?

Reply:
I'm probably missing something with the syntax so I'm not sure how you suggest to force the keys here.

Also, regarding the `*_call` methods I get `UnboundLocalError: local variable '*_call' referenced before assignment` if I move it outside `multiple`.